### DISCUSSION TITLE
Add a `--factory` parameter like in uvicorn to run FastAPI applications not only through variables but also through functions

### DISCUSSION THREAD

**Discussion by ivankarmanow** [fastapi/fastapi-cli#45] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
fastapi run --factory app.main
```

### Description

In uvicorn there is a `--factory` parameter that allows you to pass as ASGI app not only variables, but also functions that return an ASGI app. I suggest adding the same parameter to fastapi-cli or changing the process of searching for a variable with FastAPI app so that factory functions are also found

### Operating System

Linux, Windows

### Operating System Details

No details

### Python Version

3.12

**Comment by adamhadani** [comment]
👍  this will be great to have.
